public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -0,0 +1,740 @@
+            title: embeddable.vis.title,
+            pluginResourceId: response.response.id,
+            visId: embeddable.vis.id,
+            savedObjectType: 'visualization',
+            visLayerExpressionFn: fn,
+          } as ISavedAugmentVis;
+
+          const savedObject = await createAugmentVisSavedObject(
+            savedObjectToCreate
+          );
+
+          const saveObjectResponse = await savedObject.save({});
+        })
+        .catch((err: any) => {
+          dispatch(getDetectorCount()).then((response: any) => {
+            const totalDetectors = get(response, 'response.count', 0);
+            if (totalDetectors === MAX_DETECTORS) {
+              notifications.toasts.addDanger(
+                'Cannot create detector - limit of ' +
+                  MAX_DETECTORS +
+                  ' detectors reached'
+              );
+            } else {
+              notifications.toasts.addDanger(
+                prettifyErrorMessage(
+                  getErrorMessage(
+                    err,
+                    'There was a problem creating the detector'
+                  )
+                )
+              );
+            }
+          });
+        });
+      closeFlyout();
+    } catch (e) {
+    } finally {
+      formikProps.setSubmitting(false);
+    }
+  };
+
+  const initialDetectorValue = {
+    name: detectorNameFromVis,
+    index: [{ label: embeddable.vis.data.aggs.indexPattern.title }],
+    timeField: embeddable.vis.data.indexPattern.timeFieldName,
+    interval: intervalValue,
+    windowDelay: delayValue,
+    shingleSize: 8,
+    filterQuery: { match_all: {} },
+    description: '',
+    resultIndex: undefined,
+    filters: [],
+    featureList: visFeatureListToFormik(featureList),
+    categoryFieldEnabled: false,
+    realTime: true,
+    historical: false,
+  };
+
+  function formikToDetectorName(title) {
+    const detectorName =
+      title +
+      '_anomaly_detector_' +
+      Math.floor(100000 + Math.random() * 900000);
+    detectorName.replace(/[^a-zA-Z0-9-_]/g, '_');
+    return detectorName;
+  }
+
+  function visAggregationToFormik(value) {
+    return [
+      {
+        label: value.params.field.name,
+        type: 'number',
+      },
+    ];
+  }
+
+  function visAggregationQueryToFormik(value) {
+    return {
+      [snakeCase(getFeatureNameFromVisParams(value.id))]: {
+        sum: { field: value.params.field.name },
+      },
+    };
+  }
+
+  function visFeatureListToFormik(featureList): FeaturesFormikValues[] {
+    return featureList.map((feature) => {
+      return {
+        featureId: feature.id,
+        featureName: getFeatureNameFromVisParams(feature.id),
+        featureEnabled: true,
+        featureType: FEATURE_TYPE.SIMPLE,
+        importance: 1,
+        newFeature: false,
+        aggregationBy: 'sum',
+        aggregationOf: visAggregationToFormik(feature),
+        aggregationQuery: JSON.stringify(visAggregationQueryToFormik(feature)),
+      };
+    });
+  }
```

Comment:
Recommend to move these to a helpers.ts file or something similar to (1) declutter this file a bit and (2) be consistent with the rest of the plugin. Otherwise I am ok with these particular conversion functions, they all make sense to me why we can't use existing ones for these particular fields

Reply:
moved helper functions to a helper file